packages/grid/_modules_/grid/components/GridPagination.tsx

```diff
@@ -43,13 +43,12 @@ export const GridPagination = React.forwardRef<
 >(function GridPagination(props, ref) {
   const classes = useStyles();
   const apiRef = useGridApiContext();
+  const rootProps = React.useContext(GridRootPropsContext)!;
```

Comment:
Can we get rid of the `!` at the end?

Reply:
```ts
const GridRootPropsContext = React.createContext<GridComponentProps | undefined>(undefined);
```

We could do the following :

```
const useGridRootProps = () => {
  const contextValue = React.useContext(GridRootPropsContext)

  if (!contextValue) {
    throw new Error('useGridRootProps should only be used inside the DataGrid / XGrid component
  }
}
```

Or remove the `undefined` by passing the default props :

```ts
const GridRootPropsContext = React.createContext<GridComponentProps>({ ...DEFAULT_GRID_OPTIONS, rows: [], columns: [] });
```